File: lib/ui/components/ultra_pro_chip.dart
import 'package:flutter/material.dart';

enum UltraProChipVariant { assist, filter }

class UltraProChip extends StatelessWidget {
  final String label;
  final UltraProChipVariant variant;
  final bool selected;
  final VoidCallback onPressed;

  const UltraProChip({
    super.key,
    required this.label,
    required this.onPressed,
    this.variant = UltraProChipVariant.assist,
    this.selected = false,
  });

  @override
  Widget build(BuildContext context) {
    switch (variant) {
      case UltraProChipVariant.assist:
        return ActionChip(
          label: Text(label),
          onPressed: onPressed,
        );
      case UltraProChipVariant.filter:
        return FilterChip(
          label: Text(label),
          selected: selected,
          onSelected: (_) => onPressed(),
        );
    }
  }
}

File: android/ui/components/UltraProChip.kt
 package com.ultrapro.ui.components

import androidx.compose.material3.AssistChip
import androidx.compose.material3.FilterChip
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

enum class UltraProChipVariant { Assist, Filter }

@Composable
fun UltraProChip(
    label: String,
    variant: UltraProChipVariant = UltraProChipVariant.Assist,
    selected: Boolean = false,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    when (variant) {
        UltraProChipVariant.Assist -> {
            AssistChip(
                onClick = onClick,
                label = { Text(label) },
                modifier = modifier
            )
        }

        UltraProChipVariant.Filter -> {
            FilterChip(
                selected = selected,
                onClick = onClick,
                label = { Text(label) },
                modifier = modifier
            )
        }
    }
}
